File: python/system_design/database_sharding.py
class Shard: __init__(self): self.data={}
class ShardManager:
    def __init__(self, n): self.shards=[Shard() for _ in range(n)]; self.n=n
    def get_shard(self, k): return self.shards[hash(k)%self.n]
    def put(self, k, v): self.get_shard(k).data[k]=v
    def get(self, k): return self.get_shard(k).data.get(k)
if __name__ == "__main__": m=ShardManager(3); m.put("x",1); print(m.get("x"))

File: java/system_design/DatabaseSharding.java
import java.util.*;
public class DatabaseSharding{class Shard{Map<String,Object>data=new HashMap<>();}Shard[]shards;DatabaseSharding(int n){shards=new Shard[n];for(int i=0;i<n;i++)shards[i]=new Shard();}void put(String k,Object v){shards[k.hashCode()%shards.length].data.put(k,v);}public static void main(String[]a){DatabaseSharding ds=new DatabaseSharding(3);ds.put("x",1);}}
